spacy/language.py

```diff
@@ -159,7 +159,8 @@ def __init__(
         self.vocab: Vocab = vocab
         if self.lang is None:
             self.lang = self.vocab.lang
-        self.pipeline = []
+        self.components = []
```

Comment:
Arguably we should make the data `self._components`, which would allow us to make `components` a property that returns a tuple. This would "protect" components from modification.

I'm not really decided about whether this is good? It's a bit...non-free-spirited, but I think it might be better overall.

Reply:
I guess making it a property would be more consistent – although, `self.pipeline` also worked this way before.

Not sure we should return a tuple, though, as this would make it inconsistent with all the other properties that all return lists.